### ISSUE TITLE
hideChanFromHB not working for iFan02

### ISSUE THREAD

**Issue by JangoBritt** [bwp91/homebridge-ewelink#112]
I have an iFan02, which I've just set up with this plugin.

The device is reporting correctly in HomeKit, with the three-speed fan on Fanv2 type, and a separate Light accessory.

Both Fan and Light accessory report as having the serial number 10004bb450SWX in the Home app.

I'm trying to hide the Light accessory, because it has a Hue bulb in it, but it still appears in the Home App.

I've tried both hiding 10004bb450SW0, and 10004bb450SWX, but the light device stays in Home app.

**Comment by bwp91** [comment]
Hi,
Hiding the light of the iFan accessory is something that I haven't implemented in the plugin.
Basically because no one has requested it so far.
The `hideChanFromHB` configuration option only applies to generic multi-switch devices (for example the Sonoff 4CH), the idea being that empty/unused channels can be hidden.
So basically what you are trying to do is impossible in the plugin's current state. Having said that, I can definitely look into hiding the fan-light. I'll keep you posted!

**Comment by JangoBritt** [comment]
Thanks Ben. I really love what you’ve done with the plug-in since you took it over. 

The iFan02 really works flawlessly as it should. It’s by no means a showstopper to have the light switch exposed... but would be really handy to have the option to hide it, as a lot of HA users would go down the bulb route rather than the switch route

**Comment by bwp91** [comment]
~~Do you have an iFan you could share to my ewelink account?~~
Obviously you have an iFan :)
Would you be happy to share it to my ewelink account? So it's easier for me to test?

**Comment by bwp91** [comment]
@JangoBritt please try updating to the latest beta version as per [these instructions](https://github.com/bwp91/homebridge-ewelink/wiki/Beta-Version).
You should see an option in the plugin settings (Optional-Device Specific) to called "Hide Fan Lights". Please enter the device ID of your fan (10004bb450) in that box and Save. Then restart Homebridge. The light should be removed :)
Let me know how you get on.

**Comment by JangoBritt** [comment]
I installed the beta version, but it didn't seem to change the settings options visible on my HOOBS interface... do I need to manually add the setting in the config.json?

**Comment by bwp91** [comment]
It should be there. But HOOBS 😅
Try adding this to the config.json:
`"hideLightFromFan": "10004bb450",`

**Comment by JangoBritt** [comment]
no, still appears in HomeKit...

My version still appears as 3.70,  and see that beta is 3.71.. I went to try and install the beta branch again, and noticed a warning from NPM.

npm WARN notsup Unsupported engine for homebridge-ewelink@3.7.1-3: wanted: {"homebridge":">=1.1.0","node":">=12.19.0"} (current: {"node":"12.18.2","npm":"6.14.5"})
npm WARN notsup Not compatible with your version of node/npm: homebridge-ewelink@3.7.1-3
npm WARN notsup Unsupported engine for homebridge-lib@4.7.16: wanted: {"homebridge":"^1.1.0","node":"^12.19.0"} (current: {"node":"12.18.2","npm":"6.14.5"})
npm WARN notsup Not compatible with your version of node/npm: homebridge-lib@4.7.16

seems I need to do some patching, but I'm surprised that the master branch installed yesterday... unless you updated something else with beta, too?

**Comment by JangoBritt** [comment]
incredible work Ben, thank you!

it works perfectly! :-)

**Comment by bwp91** [comment]
Amazing glad it's working for you!
One small thing I encountered:
When updating the iFan device, an "on" or "off" needs to be sent to all four channels every time one of them is controlled.
So for example if changing the speed of the fan, an "on" or "off" still needs to be sent to the light channel.
If hiding the light, I have set the light channel to stay "on" all the time, as I assume if you have a Hue bulb, then this can be turned off itself even if the channel is still "on".
Hope this makes sense!
